In [2]:
import datetime              as dt
import numpy                 as np
import pandas                as pd
import matplotlib.pyplot     as plt
import seaborn               as sns

In [3]:
df = pd.read_csv('../ETL/data/product_type.csv')
df.head(5)

C:\Anaconda\envs\dynamic-pricing\lib\site-packages\IPython\core\interactiveshell.py:3062: DtypeWarning: Columns (4) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


odin                          product_type  \
0  O1657A805FB85C4E76BDEF8E5B9037492D                               Avental   
1  O14121ABC7E11848C58BE87C91943F88B6                               Avental   
2  O13E8792B144BF4F44824BC80017AA7A47             Teste de condição da água   
3  O19E2B76CEE87F48E3AFB32017EFE29993  Estabilizador biológico para aquário   
4  O1C5805AEDB9624FBDB43D7A4DECABA0F2  Estabilizador biológico para aquário   

  product_type_group identifier_type      value  
0          vestuario            gtin  110010494  
1          vestuario            gtin  110010500  
2            limpeza            gtin  116001007  
3            petshop            gtin  116006309  
4            petshop            gtin  116006507

In [4]:
df.shape

(750296, 5)

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 750296 entries, 0 to 750295
Data columns (total 5 columns):
 #   Column              Non-Null Count   Dtype 
---  ------              --------------   ----- 
 0   odin                750296 non-null  object
 1   product_type        750296 non-null  object
 2   product_type_group  750296 non-null  object
 3   identifier_type     750296 non-null  object
 4   value               750296 non-null  object
dtypes: object(5)
memory usage: 28.6+ MB


In [6]:
df.describe()

odin product_type  \
count                               750296       750296   
unique                              750295         2592   
top     O1473706EB02FE4E3193E6E83BA48B0488       Quadro   
freq                                     2        45857   

           product_type_group identifier_type        value  
count                  750296          750296       750296  
unique                     52               2       749039  
top     utilidades_domesticas            gtin  88300101382  
freq                    91192          750295            3

We can see from the table above that we have 2592 product types within 52 groups. Also, the top product type is 'Quadro' and the top group is 'utilidades domésticas'

In [8]:
df['product_type_group'].unique()

array(['vestuario', 'limpeza', 'petshop', 'jardinagem',
       'papelaria_escritorio', 'suplementos', 'utilidades_domesticas',
       'eletronicos', 'decoracao', 'cosmeticos', 'moveis',
       'casa_construcao', 'acessorios_eletrodomesticos',
       'cama_mesa_banho', 'instrumentos_musicais', 'esporte_lazer',
       'brinquedo', 'automotivos', 'alimentos', 'beleza_saude', 'joias',
       'acessorios', 'informatica', 'ferramentas',
       'acessorios_eletronicos', 'livro', 'games', 'eletroportateis',
       'bebe_cia', 'cine_foto', 'artesanato', 'beleza_cosmeticos',
       'eletrodomesticos', 'bebidas', 'hospitalar', 'calcados',
       'fantasia', 'industrial', 'higiene_saude', 'filmes_musica',
       'artigos_festa', 'acessorios_moveis', 'mercado', 'ar_condicionado',
       'ar_ventilacao', 'telefonia', 'acessorios_informatica',
       'acessorios_eletroportateis', 'sex_shop', 'camera_foto',
       'arte_artesanato', 'fitness'], dtype=object)